api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/dsd/NextAndPrevDatesCalculation.java

```diff
@@ -62,21 +62,27 @@ public CalculationResultMap evaluate(
       while (iterator.hasNext()) {
         Obs obs = iterator.next();
         if (obs.getVoided()
+            || obs.getValueDatetime() == null
+            || obs.getValueDate() == null
             || obs.getEncounter() == null
             || (obs.getEncounter() != null
                 && (obs.getEncounter().getEncounterId() != lastEncounter.getEncounterId()))) {
           iterator.remove();
         }
       }
+
       Collections.sort(
           returnVisitList,
           new Comparator<Obs>() {
             @Override
             public int compare(Obs obs, Obs t1) {
-              return t1.getValueDate().compareTo(obs.getValueDatetime());
+              int comparisonValue = 1;
+              if (t1.getValueDate() != null && obs.getValueDatetime() != null) {
```

Comment:
we can remove this condition since we are removing obs with null dates in the while loop above. No need to do this twice

Reply:
This is removed @nthfloor was just trying out something and forgot to remove.